### ISSUE TITLE
mean of y_cls  and y_cls[i, T]

### ISSUE THREAD

**Issue by upnana** [hyk1996/Single-Human-Parsing-LIP#3]
What‘s the ’mean of y_cls  and y_cls[i, T]，what they represent and what is its value

**Comment by Yukun-Huang** [comment]
You can find the explanation from [code comments](https://github.com/hyk1996/Single-Human-Parsing-LIP/blob/a83161575723728ae05dc052832c60d835dffcdb/train.py#L79).
```
    y_cls - batch of 1D tensors of dimensionality N: N total number of classes,
    y_cls[i, T] = 1 if class T is present in image i, 0 otherwise
```
"y_cls" represents which categories appear in the image and it is for calculating the auxiliary loss. More details can be found in paper "Pyramid Scene Parsing Network".

**Comment by upnana** [comment]
> You can find the explanation from [code comments](https://github.com/hyk1996/Single-Human-Parsing-LIP/blob/a83161575723728ae05dc052832c60d835dffcdb/train.py#L79).
> 
> ```
>     y_cls - batch of 1D tensors of dimensionality N: N total number of classes,
>     y_cls[i, T] = 1 if class T is present in image i, 0 otherwise
> ```
> 
> "y_cls" represents which categories appear in the image and it is for calculating the auxiliary loss. More details can be found in paper "Pyramid Scene Parsing Network

**Comment by upnana** [comment]
thank you very much   the upsample size is set according to the paper？

**Comment by Yukun-Huang** [comment]
Most settings are following this implementation [pspnet-pytorch](https://github.com/Lextal/pspnet-pytorch), so it could be slightly different from the paper.
